safe_transaction_service/history/services/index_service.py

```diff
@@ -348,15 +327,16 @@ def reindex_master_copies(
         if not addresses:
             logger.warning("No addresses to process")
         else:
-            logger.info("Start reindexing Safe Master Copy addresses %s", addresses)
-            current_block_number = ethereum_client.current_block_number
+            logger.info("Start reindexing addresses %s", addresses)
+            current_block_number = self.ethereum_client.current_block_number
             stop_block_number = (
                 min(current_block_number, to_block_number)
                 if to_block_number
                 else current_block_number
             )
             block_number = from_block_number
             while block_number < stop_block_number:
+                block_number = min(block_number, stop_block_number)
```

Comment:
When stop_block_number could be less than block_number?

Reply:
You are right, refactored this part with a `for` instead of a `while`